### PR TITLE
Add python 3.8 support and deprecate python 3.5

### PR DESCRIPTION
**Pull request by mtreinish**
### Summary

This commit adds support for running terra under python 3.8. It does
this by adding the trove classifiers to the package metadata and adding
a test job for python 3.8 (only travis for now this can be expanded in
the future when azure pipelines adds it to their image). At the same
time this commit starts the deprecation window for python 3.5. Python
3.5 goes end of life by upstream python in Sept. 2020. We should give
our users running with python 3.5 (which is about 10% of our users based
on pypi data) fair notice that when upstream python stops supporting it
we do as well.

### Details and comments

Python's documented branch support status and EoL dates are listed here: https://devguide.python.org/#branchstatus

### Waiting on upstream releases with wheels with 3.8:

- [x] scipy: https://github.com/scipy/scipy/issues/10927
- [X] ~~numpy~~ 1.17.3 included 3.8 wheels
- [x] psutil
- [x] pywinpty (for jupyter) https://github.com/spyder-ide/pywinpty/issues/123

visualization:

- [ ] matplotlib (needs >=3.2.0 worked around with https://github.com/Qiskit/qiskit-terra/pull/3268/files#diff-2ecd42f46df2d64c986adcc845fc9933R3 )
- [X] ~~pillow~~ 6.2.1 included 3.8 wheels

### CODE REVIEW

File: qiskit/__init__.py [hunk 1]
```diff
@@ -18,8 +18,15 @@
...
 
 import pkgutil
+import sys
 import warnings
 
+if sys.version_info[0] == 3 and sys.version_info[1] == 5:
+    warnings.warn(
+        "Using Qiskit with Python 3.5 is deprecated as of the 0.11.0 release. "
+        "Support for running Qiskit with Python 3.5 will be removed at the "
+        "Python 3.5 EoL on 10/13/2019.", DeprecationWarning)
```
Comment (nonhermitian) [inline qiskit/__init__.py]: 2020 maybe

Comment (mtreinish) [reply qiskit/__init__.py]: oops sry, too many dates flying around my head this morning. One sec I'll fix it. (I got it right in the release notes at least)

**Comment by atilag** [comment]
I agree, seems legit.
I will add something similar for `Aer` soon

**Comment by mtreinish** [comment]
> I agree, seems legit.
> I will add something similar for `Aer` soon

When you do we probably should not raise a warning. Otherwise we'll end up with a scenario where you get a warning from terra, aer, ignis, aqua, and ibmq-provider :)

**Comment by mtreinish** [comment]
This is currently blocked by: https://github.com/scipy/scipy/issues/10927

**Comment by mtreinish** [comment]
We're still waiting on a scipy release: https://pypi.org/project/scipy/ the next version 1.3.2 will include the needed python 3.8 fixes. There are also the other dependencies to check too

**Comment by mtreinish** [comment]
psutil uploaded a python 38 wheel (for windows which was the only env they have any wheels for) last week, so I'm checking that one off the list

**Comment by mtreinish** [comment]
also matplotlib's rc1 for the next release 3.2.0 (which was pushed last week) has python 3.8 wheels, so as soon as that's released for real we can check it off.

**Comment by mtreinish** [comment]
Ok, scipy finally pushed a release on Saturday which includes wheels for python 3.8. We're still waiting on a matplotlib release (not an rc). But I think this is probably good enough to remove the on hold tag now since the CI job should run without blas source installed.

**Comment by mtreinish** [comment]
While this is passing I'm adding the on hold tag again, because we're not verifying osx or windows. Some of my patches on other platforms are showing issues when running with python 3.8 on those so until we can fully verify on all platforms I'm going to block things.

**Comment by mtreinish** [comment]
The current state here is that linux is working fine on python 3.8, windows is stuck waiting on more upstream package releases (the PR summary message has been updated), and osx is facing some weird multiprocessing bugs, it's causing failures on things that use `parallel_map` or execute jupyter notebooks.

**Comment by mtreinish** [comment]
I'm thinking the macOS failure might be a variant of: https://bugs.python.org/issue38501

**Comment by nonhermitian** [comment]
It does indeed look like it.

**Comment by mtreinish** [comment]
After #3491 and the last 2 commits on the PR branch it looks like everything is almost working. Only `test.python.tools.jupyter.test_notebooks.TestJupyter.test_jupyter_jobs_pbars` is failing on both osx and windows for python 3.8. On windows it fails on a kernel timeout waiting 60 seconds for the notebook to finish executing. On osx it fails with the following error message: `BrokenProcessPool: A process in the process pool was terminated abruptly while the future was running or pending.`

File: qiskit/__init__.py [hunk 2]
```diff
@@ -70,4 +71,11 @@
 from .version import _get_qiskit_versions
 
 
+if sys.version_info[0] == 3 and sys.version_info[1] == 5:
+    warnings.warn(
+        "Using Qiskit with Python 3.5 is deprecated as of the 0.11.0 release. "
```
Comment (kdk) [inline qiskit/__init__.py]:
```suggestion
        "Using Qiskit with Python 3.5 is deprecated as of the 0.12.0 release. "
```

File: releasenotes/notes/deprecate-python-3.5-5bc0aadebc79d6b5.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
+---
+issues:
+  - |
+    Running functions that use ``qiskit.tools.parallel_map()`` (for example
+    ``qiskit.execute``, ``qiskit.transpile``, and
+    ``qiskit.transpiler.passmanager.PassManager.run()``) may not work when
+    called from a script running outside of a ``if __name__ == '__main__':``
+    block when using Python 3.8 on MacOS. Other environments are uneffected by
+    this issues. This is due to changes in how parallel processes are launched
```
Comment (kdk) [inline releasenotes/notes/deprecate-python-3.5-5bc0aadebc79d6b5.yaml]:
```suggestion
    this issue. This is due to changes in how parallel processes are launched
```

File: releasenotes/notes/deprecate-python-3.5-5bc0aadebc79d6b5.yaml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    by Python 3.8 on MacOS. If ``RuntimeError`` or ``AttributeError`` are raised
+    by scripts with that are directly calling ``parallel_map()`` or when
```
Comment (kdk) [inline releasenotes/notes/deprecate-python-3.5-5bc0aadebc79d6b5.yaml]:
```suggestion
    by scripts that are directly calling ``parallel_map()`` or when
```

File: releasenotes/notes/deprecate-python-3.5-5bc0aadebc79d6b5.yaml [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    calling a function that uses it internally with Python 3.8 on MacOS
+    embedding the script calls inside ``if __name__ == '__main__':`` should
+    workaround the issues. For exmaple::
```
Comment (kdk) [inline releasenotes/notes/deprecate-python-3.5-5bc0aadebc79d6b5.yaml]:
```suggestion
    workaround the issue. For example::
```

**Pull request merged**